File: 2019.cs/Day2.cs
using System.Diagnostics;

namespace AoC;

class Day2{
    static readonly Stopwatch time = new();
    public static void Solve(string input){
        time.Start();
        Console.WriteLine($"    1: {Part1(input),15} / {Part2(input),-15}  {time.Elapsed}");
    }
    private static int Part1(string input){
        int[] nums = [.. input.Split(',').Select(x => int.Parse(x))];
        int i = 0;
        nums[1] = 12;
        nums[2] = 2;
        while(nums[i]!=99){
            switch(nums[i]){
                case 1:
                    nums[nums[i+3]] = nums[nums[i+1]]+nums[nums[i+2]];
                    i += 4;
                    break;
                case 2:
                    nums[nums[i+3]] = nums[nums[i+1]]*nums[nums[i+2]];
                    i += 4;
                    break;   
            }
        }
        return nums[0];
    }
    private static int Part2(string input){
        //takes forever too lazy to think more
        // int[] nums = [.. input.Split(',').Select(x => int.Parse(x))];
        // int i = 0, noun = 0, verb = 0;
        
        // while(nums[0]!=19690720){
        //     while(nums[i]!=99){
        //         switch(nums[i]){
        //             case 1:
        //                 nums[nums[i+3]] = nums[nums[i+1]]+nums[nums[i+2]];
        //                 i += 4;
        //                 break;
        //             case 2:
        //                 nums[nums[i+3]] = nums[nums[i+1]]*nums[nums[i+2]];
        //                 i += 4;
        //                 break;   
        //         }
        //     }
        //     verb++;
        //     if(verb==100){
        //         verb = 0;
        //         noun++;
        //     }
        // }
        
        // return 100*noun+verb;
        return 0;
    }
}

File: 2015.rb/Day2.rb
class Day2
  def self.solve(input)
    start = Time.now
    printf("    2: %15s / %-15s  #{Time.now-start}\n", Part1(input.split), Part2(input.split))
  end
  def self.Part1(input)
    sum = 0
    for i in 0..input.length-1 do
      nums = input[i].split('x', -1).map(&:to_i);
      sides = [nums[0]*nums[1],nums[0]*nums[2],nums[1]*nums[2]]
      sum += 2*sides.sum+sides.min
    end
    return sum
  end
  def self.Part2(input)
    sum = 0
    for i in 0..input.length-1 do
      nums = input[i].split('x', -1).map(&:to_i);
      sum += nums.inject(:*) + 2*nums.min(2).sum
    end
    return sum
  end
end
